app/models/finance/statement/ecf.rb

```diff
@@ -15,9 +15,16 @@ def cache_original_value!
 
   def orchestrator
     Finance::ECF::CalculationOrchestrator.new(
-      aggregator: Finance::ECF::ParticipantAggregator,
+      aggregator: aggregator,
       contract: cpd_lead_provider.lead_provider.call_off_contract,
       statement: self,
     )
   end
+
+  def aggregator
+    Finance::ECF::ParticipantAggregator.new(
+      statement: self,
+      recorder: ParticipantDeclaration::ECF.where.not(state: %w[voided]),
+    )
+  end
```

Comment:
really not a big fan of coupling the database with the business logic, passing `self` is a bit of a code smell (in both orchestrator and aggregator). 
I'd rathe keep the instantiation higher up at the controller level to avoid this two ways dependencies.

Reply:
We can extract this out to a service class and delegate. I've not done that here as the payment breakdown classes are still in flux and would rather have something throw away here.